### DISCUSSION TITLE
Add support for ElysiaJS and Bun

### DISCUSSION THREAD

**Discussion by SelfhostedPro** [vikejs/bati#210] [Integration request]
Hey, Just wanted to throw in a request for supporting [ElysiaJS](https://elysiajs.com/) and [bun](https://bun.sh). All good if not but Elysia is WinterCG and seems to be a popular pick for bun users. I'm working on getting this setup on my own in the meantime.

**Comment by rtritto** [comment]
@SelfhostedPro Currently no plan for StackBlitz to support Bun (https://github.com/stackblitz/webcontainer-core/issues/1249).

Using devDependency, the installed binary is unusable (https://github.com/vikejs/bati/issues/358).

~~The [@elysiajs/node](https://github.com/elysiajs/node-adapter) adapter can be used but it doesn't support Elysia v1 (https://github.com/elysiajs/node-adapter/issues/1).~~

**Reply by SelfhostedPro** [reply]
Thanks for letting me know. You actually replied here about an alternative that works for me. Leaving it here too in case anyone is looking.
https://github.com/elysiajs/elysia/issues/494#issuecomment-2323001217

**Reply by rtritto** [reply]
~~I tested with Elysia v1 and it works (https://github.com/elysiajs/node-adapter/issues/1#issuecomment-2380709439)~~

I added some PRs to https://github.com/elysiajs/node-adapter, waiting merge.

**Reply by rtritto** [reply]
All PRs of https://github.com/elysiajs/node-adapter are merged to https://github.com/rtritto/node-adapter/tree/develop (_develop_ branch can be installed with `github:rtritto/node-adapter#develop` instead of version).

My changes with Elysia v1:
- using Node.js directly (e.g. `tsx entry.ts`) → works
- using Vike (with `vike-node`) → ~~I get a 404 route not found; probably `vike-node` should be adapted/integrated with `@elysia/node`~~ works

Some issues and PRs (see https://github.com/elysiajs/node-adapter)

**Comment by magne4000** [comment]
Just merged [Elysia support for `universal-middleware`](https://github.com/magne4000/universal-middleware/pull/39), so support for it in Bati should come soon (without node or Stackblitz support until @rtritto PR is merged).

**Reply by magne4000** [reply]
Taking a little longer than anticipated. `universal-middleware` has now been closely integrated into `vike-node`, so it should just be a matter of using `vike-node` inside Bati now. No ETA
